### DISCUSSION TITLE
Different behaviour of subtitle visuals on "normal" videos and live videos

### DISCUSSION THREAD

**Discussion by pixlhero** [SRGSSR/pillarbox-android#533] [Q&A]
### Description of the problem

When trying to increase the size of the subtitle text with `SubtitleTextSize.Fractional` it only works on non-live videos, but with live videos there is no difference. Is that something to do with how the subtitles are defined?

We would also like to move the subtitles a bit for non-live videos, because they seem to be placed at the very bottom of the videos. This again only for non-live videos, for live videos they are placed a bit higher.
I have also seen this behaviour in the "Pillarbox Demo"-App and in the "Play RTR" App.

### Reproducibility

Always

### Steps to reproduce

In the Pillarbox Demo App, check `Lists` -> `SRF` -> `News` and then for example `Mittagsausgabe`. Enable the subtitles and they are at the very bottom and quite small. The size should be able to be changed with `SubtitleTextSize.Fractional`, not their position though. (I have not tested this in this app, but in ours.)

In the Pillarbox Demo App, check `Lists` -> `TV Livestreams` -> `SRF` -> `SRF 1 live`. Enable the subtitles and they are a bit higher up and also a bit larger. `SubtitleTextSize.Fractional` should NOT be able to change this (I have not tested this in this app, but in ours.)

### Library version

1.0.0

### Operating system

Android 10

### Is there an existing issue for this?

- [X] I have searched existing issues and found no similar bug report.

**Comment by StaehliJ** [comment]
Hello thanks for the reports, can you provide us the code samples where you changes the subtitles view configuration. 

For the difference between live and vod, can you also provide the vod url/urn you see the difference. Subtitles are bundled inside stream and come in various format and subtitles providers.

For the placement you could try to add padding to your `ExoPlayerSubtitleView`  but try to keep the same aspect ratio of the player surface view. So add same padding horizontally and vertically.

**Comment by pixlhero** [comment]
Thanks for the fast reply.

The code where we tried out the options. It's not final at all. I now added the padding to test your suggestion (thanks!)

``` kotlin
ExoPlayerSubtitleView(
                modifier = Modifier.padding(all = 40.dp),
                player = player,
                captionStyle = CaptionStyleCompat(
                    /* foregroundColor = */ android.graphics.Color.WHITE,
                    /* backgroundColor = */ android.graphics.Color.BLACK,
                    /* windowColor = */ android.graphics.Color.TRANSPARENT,
                    /* edgeType = */ CaptionStyleCompat.EDGE_TYPE_DROP_SHADOW,
                    /* edgeColor = */ android.graphics.Color.BLUE,
                    /* typeface = */ null
                ),
                subtitleTextSize = SubtitleTextSize.Fractional(0.1f, true),
            )
```

Non-live urn: `urn:rtr:video:81bd93e5-5a53-4823-853d-7f6c92b3a5cc`
Live urn: `urn:srf:video:c4927fcf-e1a0-0001-7edd-1ef01d441651`

In the live urn video the subtitles already come with a padding. So adding an additional padding puts them too close to the center of the video. Also the `subtitleTextSize` seems to be ignored here (but it works for the non-live video).
I also tried both options for the `ignorePadding` flag. But it didn't make a difference.

**Comment by MGaetan89** [comment]
The styling of the subtitles may come from the subtitle stream directly. That's why some videos have padding, colours, custom size, ..., and others don't.
We would recommend to keep the default styling, to avoid losing/overriding any styling defined in the subtitle stream.

However, if you really wish to customise it for any video, you can call `setApplyEmbeddedFontSizes(false)` and/or `setApplyEmbeddedStyles(false)` on the `SubtitleView`.
This is currently not possible if you use Pillarbox's `ExoPlayerSubtitleView`, but you can create a `SubtitleView` on your side, if needed.
